lib/sheets.ts

```diff
@@ -0,0 +1,175 @@
+
+    const authUrl = oAuth2Client.generateAuthUrl({
+        access_type: 'offline',
+        scope: SCOPES,
+    });
+
+}
+*/
+
+// Exported Class Spreadsheet
+//   Methods to append row
+//   Clear Sheet
+//   Read Row
+//   Read Cell
+
+export class Spreadsheet {
+
+    private spreadsheetID: string;
+    private name: string;
+
+    constructor(spreadsheetID: string, name: string) {
+        this.spreadsheetID = spreadsheetID;
+        this.name = name;
+    }
+
+    public clear() {
+        return _await(new Promise((resolve, reject) => {
+            sheets.spreadsheets.values.clear({
+                range: this.name,
+                spreadsheetId: this.spreadsheetID,
+            }, (err: any, result: any) => {
+                if (err) {
+                    // Handle error.
+                    reject(err);
+                } else {
+                    resolve(`${result.data} cells cleared.`);
+                }
+            });
+        }));
+    }
+
+    public readRow(row: number) {
+        return _await(new Promise((resolve, reject) => {
+            sheets.spreadsheets.values.get({
+                range: this.name + "!" + row + ":" + row,
+                spreadsheetId: this.spreadsheetID,
+            }, (err: any, res: any) => {
+                if (err) {
+                    reject('The API returned an error: ' + err);
+                }
+                const rows = res.data.values;
+                if (rows.length) {
+                    resolve(rows);
+                } else {
+                    reject('No data found.');
+                }
+            });
+        }));
+    }
+
+    public readCell(cell: string) {
+        return _await(new Promise((resolve, reject) => {
+            sheets.spreadsheets.values.get({
+                range: this.name + "!" + cell,
+                spreadsheetId: this.spreadsheetID,
+            }, (err: any, res: any) => {
+                if (err) {
+                    reject('The API returned an error: ' + err);
+                }
+                const rows = res.data.values;
+                if (rows.length) {
+                    resolve(rows);
+                } else {
+                    reject('No data found.');
+                }
+            });
+        }));
+    }
+
+    public appendRow(row: string[]) {
+        return _await(new Promise((resolve, reject) => {
+            const values = row;
+            const resource = {
+                values,
+            };
+            sheets.spreadsheets.values.append({
+                range: this.name,
+                resource,
+                spreadsheetId: this.spreadsheetID,
+                valueInputOption: 'RAW',
+            }, (err: any, result: any) => {
+                if (err) {
+                    // Handle error.
+                    reject(err);
+                } else {
+                    resolve(`${result.data.updates.updatedCells} cells appended.`);
```

Comment:
Putting this string inside the `resolve()` will cause this method to return that string. I think you want to just `resolve()` with no arguments, and maybe even remove the `return` on line 122, though that won't matter if the Promise resolves to `undefined`.